test/jasmine/tests/range_slider_test.js

```diff
@@ -345,6 +345,90 @@ describe('the range slider', function() {
             })
             .then(done);
         });
+
+        it('should clear traces in range plot when needed', function(done) {
```

Comment:
@alexcjohnson This commit fixes part 1

> the data is gone from the graph, it should disappear from the range slider too. If you delete any but the last trace, the range slider does update correctly, but not when you delete the last one.

of https://github.com/plotly/plotly.js/issues/1473 in a pretty ugly way, unfortunately.

Most trace clearance (i.e. deletion or `visible: false` or `type`-restyle) is handled via this [block](https://github.com/plotly/plotly.js/blob/master/src/plots/cartesian/index.js#L104-L106):

```js
plotinfo.plot.selectAll('g:not(.scatterlayer)').selectAll('g.trace').remove();
```
except for scatter, heatmap and contour traces. That is, part 1 of #1473 actually _only_ affected scatter, heatmap and contour traces.

Scatter traces are exempted from that `selectAll('').remove()` block since @rreusser's animation PR to smooth out their data updates. They are cleared [here](https://github.com/plotly/plotly.js/blob/88ccce462bea3b38eeb4995937ddfed210b06acc/src/plots/cartesian/index.js#L150). Contour and heatmap groups don't always match one-to-one with trace items (e.g. a contour trace can have a heatmap fill group) so we use their `uid` to tell which trace to clear [here](https://github.com/plotly/plotly.js/blob/88ccce462bea3b38eeb4995937ddfed210b06acc/src/plot_api/plot_api.js#L291), [here](https://github.com/plotly/plotly.js/blob/88ccce462bea3b38eeb4995937ddfed210b06acc/src/plots/plots.js#L581) [here](https://github.com/plotly/plotly.js/blob/88ccce462bea3b38eeb4995937ddfed210b06acc/src/traces/heatmap/plot.js#L38) and [here](https://github.com/plotly/plotly.js/blob/88ccce462bea3b38eeb4995937ddfed210b06acc/src/traces/contour/plot.js#L59).

In brief, this commit ensure that every special `.remove()` call mentioned in the previous paragraph clear traces in the range slider containers as well.